Content/posts/team-restructure.md

```diff
@@ -0,0 +1,77 @@
+
+![Team Structure diagram. A discovery team with arrows for Features 1, 2 and 3 pointing at three other teams: backend, mobile and platform. The discovery team contains a tech lead, product managers, business analysts, performance analyst, user researcher and designers. The backend team contains a tech lead, typescript developers and automation QA. The mobile team contains a tech lead, Android developers, iOS developers, automation QA and manual QA. The platform team contains a tech lead and platform developers.](../../Images/team-restructure-before.png)
+*Our Old Team Structure — Teams are split by discipline*
+
+When there are more members of each specialism (i.e. Android developers) in a larger mobile team, we try and give them unrelated pieces of work to avoid merge conflicts in the code that is being written.
+This means there are multiple features or pieces of work being delivered in parallel by the team.
+Unless we clearly assign team members to a particular feature, this can result in a lot of communication overheads and redudancy.
+For example, the whole team often end up invited to meetings for **every single** feature when only a single developer is needed to provide guidance.
+
+Alternatively, it may be the tech lead or a single individual that takes on the responsibility in all meetings, which results in a very top-down hierarchy where the team as a whole does not feel empowered.
+What’s more, this can result in expert overload where the team accumulates more knowledge because they are at the centre of everything, resulting in them becoming indispensible in all situations.
+This is a vicious cycle because as they are now indispensible and they will gain even more knowledge — until they burn out and the team comes to a standstill.
+
+> _"Business people and developers must work together daily throughout the project."_
+> 
+> — Principles behind the Agile Manifesto, [agilemanifesto.org](http://agilemanifesto.org/principles.html)
+
+While agile principles state that business people and developers must work together daily, our old team structure did not encourage this behaviour.
+Since product managers and business analysts did not sit inside our teams, they were not required to attend our regular scrum ceremonies including the daily stand-up.
+We also encountered the similar communication barriers here, where it was not always clear which product manager or business analyst was responsible for a given feature.
+
+Finally, since all our team members were constantly context switching between different features, it was difficult for them to focus for long enough to gain a deep understanding of each of them.
+When this happens, we see the [Law of triviality](https://en.wikipedia.org/wiki/Law_of_triviality) begin to take effect.
```

Comment:
I don't think bike shedding is the only downside of this kind of context-switching? I think it also results in features not being completely "finished" because there's always something else to jump onto, plus the known impact of context-switching on well-being and productivity.

Reply:
Thanks 😄 I've tried to flesh out the context switching section a bit more, would be amazing if you get chance to take another look, @elishkaflint !